mayavi/core/scene.py

```diff
@@ -198,6 +200,11 @@ def tno_drop_object(self, node, dropped_object):
     ######################################################################
     # Non-public interface
     ######################################################################
+    def _closed_fired(self):
+        self.scene = None
+        self.menu_helper.object = None
+        self.parent.closed = True
```

Comment:
This should really never call engine's closed as that will remove/clear all viewer refs etc. in the engine.  Instead engine's remove_scene method could be called.  However, this is tricky as if the TVTK Scene plugin is used, it automatically takes care of removing the scenes from the engine.  I think this new event "closed" is not needed.  Let us discuss this.